engine/src/main/java/org/destinationsol/world/generators/PlanetGenerator.java

```diff
@@ -15,25 +15,244 @@
+    private boolean isInFirstSolarSystem;
+    private boolean isInnerPlanet;
+    private PlanetConfig planetConfig;
+    private float groundHeight;
     //distance from ground of planet to atmosphere
-    float atmosphereHeight;
-    float angleInSolarSystem;
-    float distanceFromSolarSystemCenter;
+    private float atmosphereHeight;
+    private float angleInSolarSystem;
+    private float distanceFromSolarSystemCenter;
+    private float initialPlanetAngle;
+    //This is the speed the Planet orbits around the SolarSystem at
+    private float orbitSolarSystemSpeed;
+    //This is the speed the Planet spins around it's axis at
+    private float planetRotationSpeed;
+
+    private PlanetConfigs planetConfigManager;
+
+    /**
+     * This method sets the planet variable for the PlanetGenerator. It should be called at the end of the build() method,
+     * after the settings of the planet have been set.
+     */
+    protected void createPlanet() {
+        planet = new Planet(getSolarSystemPosition(), getAngleInSolarSystem(), getDistanceFromSolarSystemCenter(),
+                getInitialPlanetAngle(), orbitSolarSystemSpeed, planetRotationSpeed, getGroundHeight(),
+                false, getPlanetConfig(), name);
+    }
+
+    /**
+     * The gets a planet config with the difficulty set in the way it is by default. If a planet is not an inner planet
+     * (less than half the SolarSystem radius from the center of the system), and not in the starting system, is in
+     * the starting system, it will be easy. If the Planet is in the inner part of its system, and it is not the first
+     * system, it will be a hard planet. Otherwise, it will be a medium planet.
+     * @return the PlanetConfig with the correct difficulty
+     */
+    protected PlanetConfig getPlanetConfigDefaultSettings() {
+        return getPlanetConfigManager().getRandom(!getIsInnerPlanet() && getIsInFirstSolarSystem(),
+                getIsInnerPlanet() && !getIsInFirstSolarSystem());
+    }
+
+    /**
+     * This calculates the proper speed for a planet to orbit around the center of the SolarSystem. It is randomized
+     * slightly, so not all planets orbit at the same speed.
+     * @return float representing orbit speed
+     */
+    protected float calculateDefaultPlanetOrbitSpeed() {
+        return SolMath.arcToAngle(PLANET_SPEED, getDistanceFromSolarSystemCenter()) * SolMath.toInt(SolRandom.seededTest(.5f));
+    }
+
+    /**
+     * This calculates the proper speed for a planet to rotate around its axis. It is randomized slightly, so not all
+     * planets rotate at the same speed.
+     * @return float representing rotation speed
+     */
+    protected float calculateDefaultPlanetRotationSpeed() {
+        return SolMath.arcToAngle(PLANET_GROUND_SPEED, getGroundHeight()) * SolMath.toInt(SolRandom.seededTest(.5f));
+    }
+
+    /**
+     * This method modifies the density of low orbit ships that generate on the Planet. The density represents how
```

Comment:
I wonder if "density" here is the right word, another possible refactoring that could be bigger than this PR / do sometime later. When I read density I think a desired target state of x ships per y area. But is it really more like a _probability_ to spawn ships that is based on a desired target density? Generally spawn chances are thought of in probabilities, although I can see how a target "coverage" of sorts could be a goal. *If* I'm even reading that right - you could have logic that would go past a target "density" depending on how it was written.